crates/scmi/src/devices/iio.rs

```diff
@@ -0,0 +1,851 @@
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_anglevel",
+        unit: scmi::SENSOR_UNIT_RADIANS_PER_SECOND,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_concentration",
+        unit: scmi::SENSOR_UNIT_PERCENTAGE,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_current",
+        unit: scmi::SENSOR_UNIT_AMPS,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_capacitance",
+        unit: scmi::SENSOR_UNIT_FARADS,
+        unit_exponent: -9,
+    },
+    UnitMapping {
+        channel: "in_distance",
+        unit: scmi::SENSOR_UNIT_METERS,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_electricalconductivity",
+        unit: scmi::SENSOR_UNIT_SIEMENS, // per meter
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_energy",
+        unit: scmi::SENSOR_UNIT_JOULS,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_gravity",
+        unit: scmi::SENSOR_UNIT_METERS_PER_SECOND_SQUARED,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_humidityrelative",
+        unit: scmi::SENSOR_UNIT_PERCENTAGE,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_illuminance",
+        unit: scmi::SENSOR_UNIT_LUX,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_magn",
+        unit: scmi::SENSOR_UNIT_GAUSS,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_ph",
+        unit: scmi::SENSOR_UNIT_UNSPECIFIED, // SCMI doesn't define pH
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_positionrelative",
+        unit: scmi::SENSOR_UNIT_PERCENTAGE,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_power",
+        unit: scmi::SENSOR_UNIT_WATTS,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_pressure",
+        unit: scmi::SENSOR_UNIT_PASCALS,
+        unit_exponent: 3,
+    },
+    UnitMapping {
+        channel: "in_proximity",
+        unit: scmi::SENSOR_UNIT_METERS,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_resistance",
+        unit: scmi::SENSOR_UNIT_OHMS,
+        unit_exponent: 0,
+    },
+    UnitMapping {
+        channel: "in_temp",
+        unit: scmi::SENSOR_UNIT_DEGREES_C,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_velocity_sqrt(x^2+y^2+z^2)",
+        unit: scmi::SENSOR_UNIT_METERS_PER_SECOND,
+        unit_exponent: -3,
+    },
+    UnitMapping {
+        channel: "in_voltage",
+        unit: scmi::SENSOR_UNIT_VOLTS,
+        unit_exponent: -3,
+    },
+];
+
+/// Representation of an IIO channel axis.
+///
+/// Used also for scalar values.
+#[derive(PartialEq, Debug)]
+struct Axis {
+    /// Full sysfs path to the axis value file stripped of "_raw".
+    path: String, // without "_raw" suffix
```

Comment:
Should be a PathBuf if it is an actual path?

Reply:
I think I had a reason to use String rather than PathBuf here but can't remember why exactly. Probably because it's used more in string contexts and there is no important reason to use PathBuf here.

Reply:
I changed it to OsString (not sure whether for better or worse).